chemprop/uncertainty/uncertainty_calibrator.py

```diff
@@ -377,36 +405,44 @@ def calibrate(self):
         individual_vars = np.array(
             self.calibration_predictor.get_individual_vars()
         )  # shape(models, data, tasks)
-        targets = np.array(self.calibration_data.targets())
-        errors = uncal_preds - targets
+        targets = np.array(self.calibration_data.targets(), dtype=float)
+        mask = np.array(self.calibration_data.mask(), dtype=bool)
+        self.var_weighting = np.zeros([self.num_models, targets.shape[1]])  # shape(models, tasks)
 
-        def objective(scaler_values: np.ndarray):
-            scaler_values = np.reshape(softmax(scaler_values), [-1, 1, 1])
-            scaled_vars = np.sum(
-                individual_vars * scaler_values, axis=0, keepdims=False
-            )
-            nll = np.log(2 * np.pi * scaled_vars) / 2 + (errors) ** 2 / (
-                2 * scaled_vars
-            )
-            nll = np.sum(nll, axis=0)
-            return nll
+        for i in range(targets.shape[1]):
+            task_mask = mask[:, i]
+            task_targets = targets[task_mask, i]
+            task_preds = uncal_preds[task_mask, i]
+            task_ind_vars = individual_vars[:, task_mask, i]
+            task_errors = task_preds - task_targets
+
+            def objective(scaler_values: np.ndarray):
+                scaler_values = np.reshape(softmax(scaler_values), [-1, 1])  # (models, 1)
+                scaled_vars = np.sum(
+                    task_ind_vars * scaler_values, axis=0, keepdims=False
+                )  # (data, tasks)
```

Comment:
```suggestion
                )  # (data, 1)
```
In this case, this should be modified to (data, 1)